### PR TITLE
CP-2387 - Integration tests examples

### PR DESCRIPTION
**Pull request by phpinhei-te**
Since 2.0.0 hasnt been released yet, I didnt include all changes, so the tests do not run, but they pass, trust me:
image

### CODE REVIEW

File: endpoint-test-results/src/test/java/com/thousandeyes/sdk/endpoint/tests/results/LocalNetworkEndpointTestResultsApiTest.java [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+     * <p>
+     * Returns detailed data of a local network topology. 
+     *
+     * @throws JsonProcessingException if the deserialization fails
+     */
+    @Test
+    public void getLocalNetworksTestResultsTopologyRequestAndResponseDeserializationTest()
+            throws JsonProcessingException, ApiException
+    {
+        String networkTopologyId = "00160:39c518560de9:1491651900:236e6f18";
```
Comment (phpinhei-te) [inline endpoint-test-results/src/test/java/com/thousandeyes/sdk/endpoint/tests/results/LocalNetworkEndpointTestResultsApiTest.java:307]: This is the reason why we need to encode the ids

File: endpoint-test-results/src/test/java/com/thousandeyes/sdk/endpoint/tests/results/NetworkEndpointScheduledTestResultsApiTest.java [hunk 2]
```diff
@@ -0,0 +1,1442 @@
...
+        String contentType = "application/json";
+        PathVisDetailEndpointTestResults mappedResponse = 
+                mapper.readValue(responseBodyJson, PathVisDetailEndpointTestResults.class);
+        assertNotNull(mappedResponse);
+
+        String path = "/endpoint/test-results/scheduled-tests/{testId}/path-vis/agent/{agentId}/round/{roundId}";
+        stubFor(get(urlPathTemplate(path))
+                        .withPathParam("testId", equalTo(URLEncoder.encode(testId, StandardCharsets.UTF_8)))
+                        .withPathParam("agentId", equalTo(URLEncoder.encode(agentId, StandardCharsets.UTF_8)))
+                        .withPathParam("roundId", equalTo(URLEncoder.encode(roundId, StandardCharsets.UTF_8)))
```
Comment (phpinhei-te) [inline endpoint-test-results/src/test/java/com/thousandeyes/sdk/endpoint/tests/results/NetworkEndpointScheduledTestResultsApiTest.java:1104]: Example with multiple path parameters

File: endpoint-agents/src/test/java/com/thousandeyes/sdk/endpoint/agents/EndpointAgentsApiTest.java [hunk 3]
```diff
@@ -0,0 +1,1638 @@
...
+     * <p>
+     * Deletes the agent with the specified &#x60;agent_id&#x60;. 
+     *
+     * @throws JsonProcessingException if the deserialization fails
+     */
+    @Test
+    public void deleteEndpointAgentRequestAndResponseDeserializationTest()
+            throws JsonProcessingException, ApiException
+    {
+        UUID agentId = UUID.fromString("861b7557-cd57-4bbb-b648-00bddf88ef49");
```
Comment (phpinhei-te) [inline endpoint-agents/src/test/java/com/thousandeyes/sdk/endpoint/agents/EndpointAgentsApiTest.java:89]: Sometimes id type will be UUID, and this is actually the example coming from the generator, with code and everything

File: endpoint-agents/src/test/java/com/thousandeyes/sdk/endpoint/agents/EndpointAgentsApiTest.java [hunk 4]
```diff
@@ -0,0 +1,1638 @@
...
+                                  """;
+        int statusCode = 200;
+        String contentType = "application/json";
+        EndpointAgent mappedResponse = 
+                mapper.readValue(responseBodyJson, EndpointAgent.class);
+        assertNotNull(mappedResponse);
+
+        String path = "/endpoint/agents/{agentId}/disable";
+        stubFor(post(urlPathTemplate(path))
+                        .withPathParam("agentId", equalTo(URLEncoder.encode(agentId.toString(), StandardCharsets.UTF_8)))
```
Comment (phpinhei-te) [inline endpoint-agents/src/test/java/com/thousandeyes/sdk/endpoint/agents/EndpointAgentsApiTest.java:278]: But then we need toString it to pass to wiremock stub

File: credentials/build.gradle [hunk 5]
```diff
@@ -19,6 +19,8 @@ dependencies {
     implementation "org.apache.commons:commons-lang3:$apacheCommonsLang3"
 
     testImplementation "org.junit.jupiter:junit-jupiter:$junitVersion"
+    testImplementation "org.wiremock:wiremock:3.9.1"
+    testImplementation project(":core:client-native")
```
Comment (phpinhei-te) [inline credentials/build.gradle:23]: The only two dependencies we'd need to include

File: credentials/src/test/java/com/thousandeyes/sdk/credentials/CredentialsApiTest.java [hunk 6]
```diff
@@ -99,12 +138,24 @@ public void createCredentialRequestAndResponseDeserializationTest()
...
+
+        String path = "/credentials/{id}";
+        stubFor(delete(urlPathTemplate(path))
+                        .withPathParam("id", equalTo(URLEncoder.encode(id, StandardCharsets.UTF_8)))
+                        .willReturn(aResponse()
+                                            .withHeader(AUTHORIZATION, BEARER_TOKEN)
+                                            .withStatus(statusCode)));
+
+        var apiResponse = api.deleteCredentialWithHttpInfo(id, null);
+        assertEquals(statusCode, apiResponse.getStatusCode());
```
Comment (phpinhei-te) [inline credentials/src/test/java/com/thousandeyes/sdk/credentials/CredentialsApiTest.java:158]: delete has no response, but we can still assert the status code

File: credentials/src/test/java/com/thousandeyes/sdk/credentials/CredentialsApiTest.java [hunk 7]
```diff
@@ -87,9 +112,23 @@ public void createCredentialRequestAndResponseDeserializationTest()
...
         CredentialWithoutValue mappedResponse = 
                 mapper.readValue(responseBodyJson, CredentialWithoutValue.class);
         assertNotNull(mappedResponse);
+
+        String path = "/credentials";
+        stubFor(post(urlPathTemplate(path))
+                        .withRequestBody(equalToJson(requestBodyJson))
+                        .willReturn(aResponse()
+                                            .withHeader(AUTHORIZATION, BEARER_TOKEN)
+                                            .withHeader(CONTENT_TYPE, contentType)
```
Comment (brumarqu-te) [inline credentials/src/test/java/com/thousandeyes/sdk/credentials/CredentialsApiTest.java:126]: shouldn't this 2 be on the request? 🤔

Comment (phpinhei-te) [reply credentials/src/test/java/com/thousandeyes/sdk/credentials/CredentialsApiTest.java:126]: you're absolutely right, will change that on the generator

Comment (phpinhei-te) [reply credentials/src/test/java/com/thousandeyes/sdk/credentials/CredentialsApiTest.java]: cant really put accept cause we customize it during generating, but added the request content type:
https://github.com/thousandeyes/thousandeyes-sdk-generator/pull/39/commits/5d59ca23676d175f6080a3337dacce132c8c2989

Comment (phpinhei-te) [reply credentials/src/test/java/com/thousandeyes/sdk/credentials/CredentialsApiTest.java]: Ive pushed an update to credentials test only so we can see how it looks

**Pull request closed without merging**